app/models/user.rb

```diff
@@ -6,6 +6,12 @@ class User < ApplicationRecord
   VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
 
   has_many :microposts, dependent: :destroy
+  has_many :active_relationships, class_name: "Relationship",
+    foreign_key: "follower_id", dependent: :destroy
+  has_many :passive_relationships, class_name: "Relationship",
```

Comment:
trình bày về tất cả các option của association mà ông biết cho t nhé

Reply:
Các option chung:
- `dependent:` phụ thuộc,
   + với đối số :destroy -> khi môt đối tượng bị huỷ, thì tất cả đối tượng liên kết với nó sẽ bị huỷ. Tuy nhiên nó sẽ gọi callback destroy. 
       Ví dụ: 1 user bị xoá thì toàn bộ bài viết của user đó vị xoá.
   + Với đối số :delete -> nó cũng tương tự `destroy` nhưng nó thao tác trực tiếp với database, không gọi callback
- class_name: sử dụng để cung cấp tên model khi tên của liên kết không giống tên model.
- foreign_key: Cho phép thiếp lập trực tiếp tên khoá ngoại.
- Primary_key: cho phép thiếp lập trực tiếp ten khoá chính, trong trường hợp không sử dụng trường hợp mặc định rails ( id là cột primary key)
Option chủ sở hữu: (không có belongs_to)
- thought: option này để thiếp lập 1 mối quan hệ giữa 2 model thông qua một model khác. 
- source: xác định tên liên kết nguồn cho option throught. trong trường hợp rails không thể tìm được asscoiation. Ví dụ với dòng:` has_many :followers, through: :passive_relationships, source: :follower` tức là `user` sẽ lấy được người đang follow mình qua follower trong `passive_relationship`